addons/account/models/ir_module.py

```diff
@@ -70,8 +70,9 @@ def write(self, vals):
             and (guessed := next((
                 tname
                 for tname, tvals in self.account_templates.items()
-                if tvals['country_id'] == self.env.company.country_id.id
-                or not tvals['country_id']
+                if self.env.company.country_id.id and (
```

Comment:
Shouldn't this be checked before trying to compute `guessed`?
Because like this we are iterating on the chart templates but we know in advance that we won't ever match the condition

Reply:
Indeed... this was my just a first hack I thought, if you agree I will then make it right and also update the commit message to be more explicit about the reasons.

Reply:
Yes, only instantiating the CoA if there is a country is totally fine for me.
Better to let the user chose than make a wrong suggestion that is hard to go back on

Reply:
On it...

Reply:
@william-andre done, I've tested it locally: 17.2 fails at install of l10n_syscohada, this branch doesn't. Commit message updated also.